Scheduled merges during start throw errors

Stacktrace:
```
Exception in thread "Lucene Merge Thread #0 [photo_order_v1:0]" org.apache.lucene.index.MergePolicy$MergeException: java.lang.NullPointerException
	at org.apache.lucene.index.ConcurrentMergeScheduler.handleMergeException(ConcurrentMergeScheduler.java:704)
	at org.apache.lucene.index.ConcurrentMergeScheduler$MergeThread.run(ConcurrentMergeScheduler.java:684)
Caused by: java.lang.NullPointerException
	at com.yelp.nrtsearch.server.luceneserver.NRTPrimaryNode.preCopyMergedSegmentFiles(NRTPrimaryNode.java:244)
	at org.apache.lucene.replicator.nrt.PreCopyMergedSegmentWarmer.warm(PreCopyMergedSegmentWarmer.java:58)
	at org.apache.lucene.index.IndexWriter.mergeMiddle(IndexWriter.java:4610)
	at org.apache.lucene.index.IndexWriter.merge(IndexWriter.java:4057)
	at org.apache.lucene.index.ConcurrentMergeScheduler.doMerge(ConcurrentMergeScheduler.java:625)
	at org.apache.lucene.index.ConcurrentMergeScheduler$MergeThread.run(ConcurrentMergeScheduler.java:662)
Nov 25, 2020 1:35:57 AM com.yelp.nrtsearch.server.grpc.LuceneServer$LuceneServerImpl startIndex
```
This is happening because `replicasInfos` isn't initialized until the super-class constructor call is complete, and the super-class constructor sets the `PreCopyMergedSegmentWarmer` on the writer. We can add a null check to `replicasInfos` but it might be better to set the `MergePolicy` of the `IndexWriter` to `NoMergePolicy` while `NrtPrimaryNode` is initialized.

--- a/src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java
+++ b/src/main/java/com/yelp/nrtsearch/server/luceneserver/ShardState.java
@@ -61,6 +61,9 @@ import org.apache.lucene.index.IndexReader;
 import org.apache.lucene.index.IndexWriter;
 import org.apache.lucene.index.IndexWriterConfig;
 import org.apache.lucene.index.KeepOnlyLastCommitDeletionPolicy;
+import org.apache.lucene.index.LiveIndexWriterConfig;
+import org.apache.lucene.index.MergePolicy;
+import org.apache.lucene.index.NoMergePolicy;
 import org.apache.lucene.index.PersistentSnapshotDeletionPolicy;
 import org.apache.lucene.index.SegmentInfos;
 import org.apache.lucene.index.SortedSetDocValues;
@@ -636,7 +639,11 @@ public class ShardState implements Closeable {
       writer =
           new IndexWriter(
               indexDir, indexState.getIndexWriterConfig(openMode, origIndexDir, shardOrd));
-      snapshots = (PersistentSnapshotDeletionPolicy) writer.getConfig().getIndexDeletionPolicy();
+      LiveIndexWriterConfig writerConfig = writer.getConfig();
+      MergePolicy mergePolicy = writerConfig.getMergePolicy();
+      // Disable merges while NrtPrimaryNode isn't initalized (ISSUE-210)
+      writerConfig.setMergePolicy(NoMergePolicy.INSTANCE);
+      snapshots = (PersistentSnapshotDeletionPolicy) writerConfig.getIndexDeletionPolicy();
 
       // NOTE: must do this after writer, because SDP only
       // loads its commits after writer calls .onInit:
@@ -670,6 +677,8 @@ public class ShardState implements Closeable {
                 }
               },
               verbose ? System.out : new PrintStream(OutputStream.nullOutputStream()));
+      // Enable merges
+      writerConfig.setMergePolicy(mergePolicy);
 
       // nocommit this isn't used?
       searcherManager =

